addons/account/models/account_move.py

```diff
@@ -4659,8 +4672,13 @@ def default_get(self, default_fields):
             and (self._context.get('journal_id') or self._context.get('default_journal_id')) \
             and self._context.get('default_move_type') in ('out_invoice', 'out_refund', 'in_invoice', 'in_refund', 'out_receipt', 'in_receipt'):
             # Fill missing 'account_id'.
-            journal = self.env['account.journal'].browse(self._context.get('default_journal_id') or self._context['journal_id'])
-            values['account_id'] = journal.default_account_id.id
+            if self._context.get("selected_partner_id"):
+                # Get most used account of the partner as the default.
+                account_id = self.env['account.account']._order_by_frequency_per_partner(
+                    self.env.company.id, self._context.get("selected_partner_id"), self._context.get("selected_move_type"), limit=1)[0]
```

Comment:
Maybe call the `name_get` with an empty string as a `name`?

Reply:
When creating a new invoice, the `company_id` is not set yet (=`False`) at this point :/

Reply:
You can get it from the journal, by keeping the line 4662 (before your changes)